enterprise/app/jobs/captain/documents/crawl_job.rb

```diff
@@ -2,17 +2,63 @@ class Captain::Documents::CrawlJob < ApplicationJob
   queue_as :low
 
   def perform(document)
-    if InstallationConfig.find_by(name: 'CAPTAIN_FIRECRAWL_API_KEY')&.value.present?
+    if pdf_document?(document)
```

Comment:
I would write a different job for this and keep this just for crawling the URLs. Create something like PDFExtractionJob.